File: src/stringparser/Language.java
/*
 * Copyright (c) 2015 Julien Guerinet. All rights reserved.
 */

package stringparser;

/**
 * One language that the Strings are in
 * @author Julien Guerinet
 * @version 1.0
 * @since 1.0
 */
public class Language {
    /**
     * The language Id
     */
    private String id;
    /**
     * The language path
     */
    private String path;
    /**
     * The column index of the language in the CSV file
     */
    private int columnIndex;

    /**
     * Default Constructor
     *
     * @param id   The language Id
     * @param path The language path
     */
    public Language(String id, String path){
        this.id = id;
        this.path = path;
        this.columnIndex = -1;
    }

    /* GETTERS */

    /**
     * @return The language Id
     */
    public String getId(){
        return this.id;
    }

    /**
     * @return The language path
     */
    public String getPath(){
        return this.path;
    }

    /**
     * @return The column index of this language
     */
    public int getColumnIndex(){
        return this.columnIndex;
    }

    /* SETTERS */

    /**
     * @param columnIndex The column index of this language
     */
    public void setColumnIndex(int columnIndex){
        this.columnIndex = columnIndex;
    }
}

File: src/main/java/Language.kt
/*
 * Copyright 2013-2020 Julien Guerinet
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 */

package com.guerinet.weave

import kotlinx.serialization.Required
import kotlinx.serialization.Serializable

/**
 * One language that the Strings are in
 * @author Julien Guerinet
 * @since 2.0.0
 *
 * @param id Language Id
 * @param path Path to the file for the Strings in this language
 * @param columnIndex Index of the column of this language in the CSV file (starts as -1)
 */
@Serializable
class Language(@Required val id: String, @Required val path: String, var columnIndex: Int = -1) {

    companion object {

        /** Dummy [Language]  object to use when we do not care about the language */
        val EMPTY = Language("en", "", 0)
    }
}
